src/publisher.h

```diff
@@ -0,0 +1,39 @@
+#ifndef DD_OPENTRACING_PUBLISHER_H
+#define DD_OPENTRACING_PUBLISHER_H
+
+#include <datadog/opentracing.h>
+
+#include <sstream>
+
+namespace datadog {
+namespace opentracing {
+
+struct SpanData;
+using Trace = std::unique_ptr<std::vector<std::unique_ptr<SpanData>>>;
+
+class AgentHttpPublisher : public TracePublisher {
+ public:
+  AgentHttpPublisher();
+  ~AgentHttpPublisher() override {}
+
+  // Returns the path that is used to submit HTTP requests to the agent.
+  const std::string path() override;
+  std::size_t pendingTraces() override;
+  void clearTraces() override;
+  // Returns the HTTP headers that are required for the collection of traces.
+  const std::map<std::string, std::string> headers() override;
+  // Returns the encoded payload from the collection of traces.
```

Comment:
May as well put these comments in the superclass?

Reply:
Done